### PR TITLE
[INLONG-5377][Sort] Add reporting metrics for kafka connector sink and audit SDK

### LINKED ISSUE #5377: [Feature][Sort] Add reporting metrics for kafka connector sink
### Description

Add reporting metrics for kafka connector sink

### Use case

 reporting metrics for kafka connector sink

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Oneal65**
- Fixes #5377 

### Motivation

Add reporting metrics for kafka connector sink

### Modifications

kafka connector

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Review by thesumery — APPROVED** [review]

File: inlong-sort/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/KafkaInLongOptions.java [hunk 1]
```diff
@@ -22,8 +22,8 @@
 import org.apache.flink.configuration.ConfigOptions;
 
 /** Option utils for Kafka table source sink. */
-public class KafkaOptions {
-    private KafkaOptions() {
+public class KafkaInLongOptions {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/KafkaInLongOptions.java:25]: This refactoring doesn't make sense, it is recommended to keep it as it is.

Comment (Oneal65) [reply inlong-sort/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/KafkaInLongOptions.java:25]: OK, 3q

File: inlong-sort/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/KafkaInLongOptions.java [hunk 2]
```diff
@@ -35,4 +35,10 @@ private KafkaOptions() {
                     .defaultValue(false)
                     .withDescription("Regard upsert delete as insert kind.");
 
+    public static final ConfigOption<String> INLONG_METRIC =
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/KafkaInLongOptions.java:38]: inlong_metric is defined in connector-base

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by yunqingmoswu — APPROVED** [review]
LGTM

**Pull request merged**